lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -1,4 +1,36 @@
 -- Propcore is allowed to everyone, but functions in the restrictedFunctions array will be restricted to devotee+ only
+local file_name = "cfc_propcore_whitelist"
+local whitelistTable = whitelistTable or {}
+
+if not file.Exists(file_name..".txt", "DATA") then
+    file.Write(file_name..".txt", "") --Creates new data file
+else
+    whitelistTable = util.JSONToTable(file.Read(file_name..".txt")) or {}
+end
+
+local function saveWhitelistChanges()
+    file.Write(file_name..".txt", util.TableToJSON(whitelistTable, true))
+end
+
+function addPlayerToPropcoreWhitelist(players)
+    for _, ply in pairs(players) do
+        table.insert(whitelistTable, ply:SteamID())
+    end
+
+    saveWhitelistChanges()
+end
+
+function removePlayerFromPropcoreWhitelist(players)
+    for _, ply in pairs(players) do
+        table.RemoveByValue(whitelistTable, ply:SteamID())
+    end
+
+    saveWhitelistChanges()
+end
+
+function isPlayerWhitelisted(ply)
```

Comment:
Maybe you could rename `whitelistTable` to `playerIsWhitelisted` and then you could just do:

```lua
if playerIsWhitelisted[ply:SteamID()] then
```

And adding to the whitelist would be as simple as:
```lua
playerIsWhitelisted[ply:SteamID()] = true
```

and removing:
```
playerIsWhitelisted[ply:SteamID()] = nil
```

Reply:
Not a fan of this method of formatting the table in this case... Leaves unnecessary garbage in the file as apposed to just having the players steamID's in one place. Like such: 
![image](https://user-images.githubusercontent.com/40676794/62821061-431a4780-bb34-11e9-942d-57724209df87.png)

Reply:
If I ever wanted to manually add a player to the list I just put their steamID in quotes along with a comma...

Reply:
Yes, it keeps the table simpler, but if you adopted this pattern you could actually remove more "garbage", by removing an entire function and relying on the table to do the lookups.

Reply:
it also has a performance benefit if the table ever gets large

Reply:
Fair point